= class NameError < StandardError

未定義のローカル変数や定数を使用したときに発生します。

例:

  bar
  # => NameError: undefined local variable or method `bar' for main:Object

== Class Methods

--- new(error_message = "", name = nil) -> NameError

例外オブジェクトを生成して返します。

@param error_message エラーメッセージを表す文字列です

@param name 未定義だったシンボルです

例:

  err = NameError.new("message", "foo")
  p err       # => #<NameError: message>
  p err.name  # => "foo"

== Instance Methods

--- name -> Symbol

この例外オブジェクトを発生させる原因となった
変数や定数、メソッドの名前をシンボルで返します。

例:

  begin
    foobar
  rescue NameError => err
    p err       # => #<NameError: undefined local variable or method `foobar' for main:Object>
    p err.name  # => :foobar
  end

--- to_s -> String

例外オブジェクトを文字列に変換して返します。

例:

  begin
    foobar
  rescue NameError => err
    p err       # => #<NameError: undefined local variable or method `foobar' for main:Object>
    p err.to_s  # => "undefined local variable or method `foobar' for main:Object"
  end

#@since 2.3.0
--- receiver -> object

self が発生した時のレシーバオブジェクトを返します。

--- local_variables -> [Symbol]

self が発生した時に定義されていたローカル変数名の一覧を返します。

内部での使用に限ります。
#@end
